types.go

```diff
@@ -25,32 +31,36 @@ const (
 	TemplateResourceGroupVersion = SmithResourceGroup + "/" + TemplateResourceVersion
 
 	TemplateNameLabel = TemplateResourceName + "/templateName"
-
-	ThirdPartyResourceGroupVersion = "extensions/v1beta1"
-	ThirdPartyResourcePath         = "thirdpartyresources"
-
-	AllNamespaces = ""
-	AllResources  = ""
 )
 
 type TemplateList struct {
-	TypeMeta `json:",inline"`
+	unversioned.TypeMeta `json:",inline"`
 	// Standard list metadata.
-	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
-	ListMeta `json:"metadata,omitempty"`
+	Metadata unversioned.ListMeta `json:"metadata,omitempty"`
 
 	// Items is a list of templates.
 	Items []Template `json:"items"`
 }
 
+// GetObjectKind is required to satisfy Object interface.
+func (tl *TemplateList) GetObjectKind() unversioned.ObjectKind {
+	return &tl.TypeMeta
+}
+
+// GetListMeta is required to satisfy ListMetaAccessor interface.
+func (tl *TemplateList) GetListMeta() meta.List {
+	return &tl.Metadata
+}
+
+var _ runtime.Object = &TemplateList{}
```

Comment:
how about moving these to a test ? They don't need to be executed every time we run the program.